### PR TITLE
WIP : Modularize sql query to string

### PR DESCRIPTION
**Pull request by PrateekGarg-gs**

### CODE REVIEW

File: legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/dbTestRunner/shared.pure [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      x:Any[1]| []
+   ]));
+}
+
+function meta::relational::dbTestRunner::runTestsWith(dbTestFns: Function<{DbTestConfig[1] -> String[0..1]}>[*],
+                                               dbType: DatabaseType[1],
+                                               expectedSqls: Map<Function<{DbTestConfig[1] -> String[0..1]}>, String>[0..1]
+                                               ): Boolean[1]
+{
+  let failures = $dbTestFns->map(f|
```
Comment (epsstan) [inline legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/dbTestRunner/shared.pure:43]: Minor. We are passing 2 structures into this function. An array of functions and a map of functions to SQLs. Can we pass in just the map ?

Comment (epsstan) [reply legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/dbTestRunner/shared.pure:43]: Ignore this comment. Expected SQL is optional.

File: legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/sqlQueryToString/h2.pure [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+   let selectOutsideWhen = ^GenerationState(generationSide=GenerationSide.Select, withinWhenClause=false);
+   let whereInsideWhen   = ^GenerationState(generationSide=GenerationSide.Where,  withinWhenClause=true);
+   let whereOutsideWhen  = ^GenerationState(generationSide=GenerationSide.Where,  withinWhenClause=false);
+   let allStates = [$selectInsideWhen, $selectOutsideWhen, $whereInsideWhen, $whereOutsideWhen];
+   let notSelectOutsideWhen = [$selectInsideWhen, $whereInsideWhen, $whereOutsideWhen];
+   
+  let dbType=DatabaseType.H2;
+
+  let dispatch = [
+      forDynafunction1('abs',                   [ choice1($allStates,            ^ToSql(format='abs(%s)'))]),
```
[the remark this answers is not in the document]
Comment (PrateekGarg-gs) [reply legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/sqlQueryToString/h2.pure:25]: yes corresponding functions in sqlQueryTOString will be deleted

File: legend-pure-code-compiled-core/src/main/resources/core/pure/protocol/vX_X_X/invocations/execution.pure [hunk 3]
```diff
@@ -246,3 +252,18 @@ function meta::protocols::pure::vX_X_X::invocation::execution::execute::alloyExe
    assertEq(200, $resp.statusCode, | $resp.statusCode->toString()+' \''+$resp.entity+'\'');
    $resp.entity;
 }
+
+function meta::protocols::pure::vX_X_X::invocation::execution::execute::legendExecutePlanInTestDatabase(plan:meta::pure::executionPlan::ExecutionPlan[1],dbType:meta::relational::runtime::DatabaseType[1], context:ExecutionContext[0..1], host:String[1], port:Integer[1], version:String[1], serializationKind:String[1], executionMode:String[1], extensions:meta::pure::router::extension::RouterExtension[*]):meta::pure::mapping::Result<Any|*>[1]
```
[the remark this answers is not in the document]
Comment (PrateekGarg-gs) [reply legend-pure-code-compiled-core/src/main/resources/core/pure/protocol/vX_X_X/invocations/execution.pure:256]: this function has been replaced now with executePlan() , in new PR, which is generic ( no relation to db specific stuff).

split params for readibility - was following earlier code convention in the file . Ofcourse we can refactor the file to make it readable . In fact we should add mvn checkstyle for java code and lay some guidelines for pure code

File: legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/sqlQueryToString/h2.pure [hunk 4]
```diff
@@ -0,0 +1,150 @@
+import meta::relational::functions::sqlQueryToString::*;
+import meta::relational::runtime::*;
+
+function <<access.private>> meta::relational::functions::sqlQueryToString::h2ReservedWords():String[*]
```
[the remark this answers is not in the document]
Comment (PrateekGarg-gs) [reply legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/sqlQueryToString/h2.pure:4]: yes , infact we should try to make conventions for pure code in general , maybe add 'checkstyle' to analyze pure code in PRs also

**Pull request closed without merging**